components/PriceRanges/__tests__/PriceRangesForm-test.js

```diff
@@ -0,0 +1,85 @@
+/* eslint-env mocha */
+
+import React from 'react';
+import expect from 'expect';
+import TestUtils from 'react-addons-test-utils';
+import sinon from 'sinon';
+import jsdom from 'mocha-jsdom';
+
+import expectJSX from 'expect-jsx';
+expect.extend(expectJSX);
+
+import PriceRangesForm from '../PriceRangesForm';
+
+describe('PriceRangesForm', () => {
+  var renderer;
+
+  jsdom({useEach: true});
+
+  beforeEach(() => {
+    let {createRenderer} = TestUtils;
+    renderer = createRenderer();
+  });
+
+  function render(extraProps = {}) {
+    var props = {
+      ...extraProps
+    };
+    renderer.render(<PriceRangesForm {...props} />);
+    return renderer.getRenderOutput();
+  }
+
+  context('display', () => {
+    it('should pass all css classes and labels', () => {
+      let out = render({
+        labels: {
+          currency: '$',
+          separator: 'to',
+          button: 'Go'
+        },
+        cssClasses: {
+          form: 'form',
+          label: 'label',
+          input: 'input',
+          currency: 'currency',
+          separator: 'separator',
+          button: 'button'
+        }
+      });
+      expect(out).toEqualJSX(
+        <form className="form" onSubmit={() => {}} ref="form">
+          <label className="label">
+            <span className="currency">$ </span>
+            <input className="input" ref="from" type="number" />
+          </label>
+          <span className="separator"> to </span>
+          <label className="label">
+            <span className="currency">$ </span>
+            <input className="input" ref="to" type="number" />
+          </label>
+          <button className="button" type="submit">Go</button>
+        </form>
+      );
+    });
+  });
+
+  context('submit', () => {
+    it('starts a refine on submit', () => {
+      // Given
+      let refine = sinon.spy();
+      let handleSubmitMock = sinon.spy(PriceRangesForm.prototype, 'handleSubmit');
```

Comment:
> If you have a better solution for not spying on the prototype let me know and I'll update the PR. Otherwise I'll simply fix the restore/naming.

The PR was merged no big deal but here it is how it was done before: https://github.com/algolia/instantsearch.js/pull/390/files#diff-40651fdb5ca1bd7385705422de4e7029L102



> Well... I tried but failed. <PriceRangesForm refine={refine} /> does not have an handleSubmit method so I cannot spy it. Once "rendered into document" it does have one, but it is then too late to spy on it as it has already been called. That's why I ended up spying on the prototype.

Weird 1. it does not have one, I thought we already did the same tests in other components? 2. is it really called on render? Because it should be called only on submit